specification/maps/data-plane/Search/preview/2022-02-01-preview/geocoding.json

```diff
@@ -0,0 +1,722 @@
+    },
+    "Bbox": {
+      "name": "bbox",
+      "in": "query",
+      "description": "A rectangular area on the earth defined as a bounding box object. The sides of the rectangles are defined by latitude and longitude values. When you specify this parameter, the geographical area is taken into account when computing the results of a location query.\n\nExample: lon1,lat1,lon2,lat2",
+      "required": false,
+      "type": "array",
+      "items": {
+        "type": "number",
+        "format": "double"
+      },
+      "collectionFormat": "csv",
+      "minItems": 4,
+      "maxItems": 4,
+      "x-ms-parameter-location": "method"
+    },
+    "View": {
+      "name": "view",
+      "in": "query",
+      "description": "A string that an [ISO 3166-1 Alpha-2 region/country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). This will alter Geopolitical disputed borders and labels to align with the specified user region.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "paths": {
+    "/geocode": {
+      "get": {
+        "description": "**Geocoding**\n\n**Applies to:** see pricing [tiers](https://aka.ms/AzureMapsPricingTier).\n\nIn many cases, the complete search service might be too much, for instance if you are only interested in traditional geocoding. Search can also be accessed for address look up exclusively. The geocoding is performed by hitting the geocoding endpoint with just the address or partial address in question. The geocoding search index will be queried for everything above the street level data. No POIs will be returned. Note that the geocoder is very tolerant of typos and incomplete addresses. It will also handle everything from exact street addresses or street or intersections as well as higher level geographies such as city centers, counties, states etc.\n>[!Important]\n>By using this feature, you agree to the preview legal terms. See the [Preview Supplemental Terms](https://azure.microsoft.com/en-us/support/legal/preview-supplemental-terms/) for additional details.",
+        "operationId": "Search_GetGeocoding",
+        "produces": [
+          "application/geo+json"
+        ],
+        "x-ms-examples": {
+          "Search detail address 15127 NE 24th Street, Redmond, WA": {
+            "$ref": "./examples/ForwardGeocoding.json"
+          },
+          "Search detail address 15127 NE 24th Street, Redmond, WA by addressLine": {
+            "$ref": "./examples/ForwardGeocodingByAddressLine.json"
+          },
+          "Search detail address 15127 NE 24th Street, Redmond, WA by query": {
+            "$ref": "./examples/ForwardGeocodingByQuery.json"
+          },
+          "Search landmark Empire State Building by query": {
+            "$ref": "./examples/ForwardGeocodingFindLandmarkByQuery.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../Common/preview/1.0/common.json#/parameters/ClientId"
```

Comment:
While it doesn't change the REST API, we want to move the client id to one of the last entries in the array. Tools sometimes do (and autorest in particular does) use the order of parameters to construct methods when they generate code, and we want the less often used optional parameters to be last.

Reply:
good suggestion. moved.